src/grammar/grammar.js

```diff
@@ -176,11 +176,33 @@ export default class Grammar {
    * Reads grammar file data. Supports reading `bnf`,
    * and `lex` grammars based on mode.
    */
-  static dataFromGrammarFile(grammarFile, grammarType = 'bnf') {
-    return Grammar.dataFromString(
-      fs.readFileSync(grammarFile, 'utf-8'),
-      grammarType
-    );
+  static dataFromGrammarFile(grammarFile, grammarType = 'bnf', loc = false) {
+    const grammar = fs.readFileSync(grammarFile, 'utf8');
+
+    // check if the bnf grammar contains location capture characters
+    if (grammarType === 'bnf' && !loc) {
+      const bnf = grammar
+        // lexer rules
+        .replace(/%lex[\n\s\S]*?\/lex/g, '')
+        // comments
+        .replace(/\/\*[\n\s\S]*?\*\//g, '')
+        .replace(/\/\/.*?\n/g, '')
+        //strings
+        .replace(/'(\\.|[^'\\])*'/g, '')
+        .replace(/"(\\.|[^"\\])*"/g, '')
+        .replace(/`(\\.|[^`\\])*`/g, '')
+        // module include
+        .replace(/%{[\n\s\S]*?%}/g, '');
```

Comment:
Not sure if we need to use all these `replace` calls, probably just can just do a simple regexp test for:

```js
if (/@\d+/.test(grammar) {
  // Warning
}
```

Reply:
without those replaces, the warning would be displayed even when finding @ inside strings, comments, lex rules, etc, which isn't good, for example, css uses that character a lot, for media queries and stuff, so it could be part of a lexer rule if someone tries to make a parser for that language. im sure there are more languages that use that character. 
also just using \d in the regex would not show the error when using named tokens (which i do a lot)

let me know if you want me to make that change anyway tho, after all its just a warning, not an error

Reply:
Yeah, sounds good - the only thing we need to make sure those replaces don't slow down much the handling.

Reply:
i tested it and it doesn't take even 0.015ms in a file of 10 millon characters, at least on my pc. also thats executed only once, and when generating the parser, it won't affect the parser itself at all, so i don't think it's a problem